test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -524,6 +560,163 @@ protected static IndexReader maybeWrapReaderEs(DirectoryReader reader) throws IO
         }
     }
 
+    protected AggregationBuilder createAggBuilderForTypeTest(MappedFieldType fieldType, String fieldName) {
+        return null;
+    }
+
+    protected List<ValuesSourceType> getSupportedValuesSourceTypes() {
+        return Collections.emptyList();
+    }
+
+    public void testSupportedFieldTypes() throws IOException {
+
+        MapperRegistry mapperRegistry = new IndicesModule(Collections.emptyList()).getMapperRegistry();
+        Settings settings = Settings.builder().put("index.version.created", Version.CURRENT.id).build();
+        String fieldName = "typeTestFieldName";
+        List<ValuesSourceType> supportedVSTypes = getSupportedValuesSourceTypes();
+
+        if (supportedVSTypes.isEmpty()) {
+            // If the test says it doesn't support any VStypes, it has not been converted yet so skip
+            return;
+        } else if (supportedVSTypes.contains(CoreValuesSourceType.ANY)) {
+            throw new IllegalArgumentException("Tests should not specify CoreValuesSourceType.ANY as a supported ValuesSourceType, " +
+                "but should instead list the concrete ValuesSourceTypes that are supported");
+        }
+
+        for (Map.Entry<String, Mapper.TypeParser> mappedType : mapperRegistry.getMapperParsers().entrySet()) {
+            if (TYPE_TEST_BLACKLIST.contains(mappedType.getKey())) {
+                continue;
+            }
+
+            if (mappedType.getKey().equals(ObjectMapper.CONTENT_TYPE)) {
+                // Cannot aggregate objects
+                continue;
+            }
+
+            Map<String, Object> source = new HashMap<>();
+            source.put("type", mappedType.getKey());
+            source.put("doc_values", "true");
+
+            FieldMapper mapper = null;
+            try {
+                Mapper.Builder builder = mappedType.getValue().parse(fieldName, source, new MockParserContext());
+                mapper = (FieldMapper) builder.build(new BuilderContext(settings, new ContentPath()));
+            } catch (Exception e) {
+                fail();
```

Comment:
Why not just let the exception happen here? The test will still fail, and the exception is likely to be clearer as to what went wrong than just a call to `fail()`

Reply:
Leftovers from debugging, this should not be here.  WIP :)